hera_mc/geo_handling.py

```diff
@@ -19,298 +21,397 @@
-                if show_location:
-                    if args.verbosity == 'm' or args.verbosity == 'h':
-                        print('station_name: ', a.station_name)
-                        print('\teasting: ', a.easting)
-                        print('\tnorthing: ', a.northing)
-                        print('\tlon/lat:  ', a.lon, a.lat)
-                        print('\televation: ', a.elevation)
-                        print('\tstation description ({}):  {}'.format(this_station, station_type[this_station]['Description']))
-                        print('\tever connected:  ', ever_connected)
-                        print('\tactive:  ', active)
-                        print('\tcreated:  ', cm_utils._get_displayTime(a.created_date))
-                    elif args.verbosity == 'l':
-                        print(a, this_station)
-    if show_location:
-        if not found_it and args.verbosity == 'm' or args.verbosity == 'h':
-            print(args.locate, ' not found.')
-    return found_location
-
-def get_all_locations(args):
-    db = mc.connect_to_mc_db(args)
-    with db.sessionmaker() as session:
-        stations = session.query(geo_location.GeoLocation).all()
-        connections = session.query(part_connect.Connections).all()
-        stations_new = []
+        Parameters:
+        ------------
+        args:  needed arguments to open database and set date/time
+        station_name:  string name of station
+        active_date:  astropy Time to check if active, ignores if not an instance of Time
+        """
+
+        connected_station = self.session.query(part_connect.Connections).filter(part_connect.Connections.upstream_part == station_name)
+        if connected_station.count() > 0:
+            station_connected = True
+        else:
+            station_connected = False
+        if station_connected and isinstance(active_date,Time):
+            counter = 0
+            for connection in connected_station.all():
+                connection.gps2Time()
+                if cm_utils._is_active(active_date, connection.start_date, connection.stop_date):
+                    station_connected = connection.downstream_part+':'+connection.down_part_rev
+                    counter += 1
+                else:
+                    station_connected = False
+            if counter > 1:
+                print("Error:  more than one active connection for", station_name)
+                station_connected = False
+        return station_connected
+
+
+    def find_station_of_antenna(self, antenna, query_date):
+        """
+        checks to see at which station an antenna is located
+
+        Returns False or the active station_name (must be an active station for the query_date)
+
+        Parameters:
+        ------------
+
+        antenna:  antenna number as float (why?), int, or string. If needed, it prepends the 'A'
+        query_date:  is the astropy Time for contemporary antenna
```

Comment:
That's the purpose of tests :). If it needs to be a float to support some specific use case, that case needs to be a test so that you can ensure that it always works (or verify that it can change to int if other code changes).